User statistics: add matomo javascript code snippet

nobeeakon opened the pull request:
Replaces #338

adds matomo tracker with disabled cookies and a privacy page. A link for privacy page is added in the footer.

![imagen](https://user-images.githubusercontent.com/12853324/201686656-f7876a9b-45f3-45f6-bf2d-5153d846fdfb.png)

nobeeakon commented:
Hey @neteler  could you please take a look at this one?

neteler commented:
For testing, I have deployed it on the server but the JS code isn't visible 

The pages contain

````
<script type="module" src="[https://grass.osgeo.org/plugins/tracking/matomo.js](view-source:https://grass.osgeo.org/plugins/tracking/matomo.js)"></script>
````

but it does not seem to trigger anything on https://2022.foss4g.org/matomo/ so far.

echoix commented:
> `<script type="module" src="[https://grass.osgeo.org/plugins/tracking/matomo.js](view-source:https://grass.osgeo.org/plugins/tracking/matomo.js)"></script>`

Is the `type="module"` expected here? Without that attribute, does it work? If the file is really the 10 lines of code I see in the PR, it doesn't really look like a js module. I know the module type exists, that brings back the js in browser closer to the structure I'm used in Node.js, but it started existing after I learned to do some JavaScript and HTML and I never really got to play again in this. Doesn't modules need some export or import declarations or something like that?


Also, the code block in this PR has the URL written weirdly with braces and a view-source url, but I checked on the current deployed website and it seems ok.

lucadelu commented:
> For testing, I have deployed it on the server but the JS code isn't visible
> 
> The pages contain
> 
> ```
> <script type="module" src="[https://grass.osgeo.org/plugins/tracking/matomo.js](view-source:https://grass.osgeo.org/plugins/tracking/matomo.js)"></script>
> ```
> 
> but it does not seem to trigger anything on https://2022.foss4g.org/matomo/ so far.

In [this document](https://developer.matomo.org/guides/tracking-javascript-guide) they say to add the code in the head or at the start of body not in the end of the page, could you try to move it?

neteler commented on themes/grass/layouts/partials/footer.html:
```diff
@@ -2,7 +2,7 @@
      <div class="row">
       <div class="col-md-4 text-md-left text-center">
-       {{ with .Site.Params.copyright }} <p><small>&copy; {{ replace . "{year}" now.Year }} (<a href="/about/license/">license</a>)</small></p> {{ end }}
+       {{ with .Site.Params.copyright }} <p><small>&copy; {{ replace . "{year}" now.Year }} (<a href="/about/license/">license</a>&nbsp;|&nbsp;<a href="about/privacy">privacy</a>)</small></p> {{ end }}
```
```suggestion
       {{ with .Site.Params.copyright }} <p><small>&copy; {{ replace . "{year}" now.Year }} (<a href="/about/license/">license</a>&nbsp;|&nbsp;<a href="/about/privacy">privacy</a>)</small></p> {{ end }}
```

neteler commented:
I have moved it to `themes/grass/layouts/partials/head.html` on the server but still nothing...

echoix commented:
And you don't have an Adblock extension installed on your browser, right? Just in case.

neteler commented:
Right I had :-) But still nothing (and I hope I was not the only visitor in the past hours...

![image](https://user-images.githubusercontent.com/1295172/202449177-0712b5b6-7b87-4a70-9d9b-f40926ff65ce.png)

nobeeakon commented:
In the [site](https://grass.osgeo.org/) I see it loads the required script
![imagen](https://user-images.githubusercontent.com/12853324/202710554-967114c6-3dce-4940-903e-5696f698882a.png)

Also I didn't see that weird URL 
![imagen](https://user-images.githubusercontent.com/12853324/202712435-e5aa6bc9-48ba-49bf-97b5-199c3a5fbbfa.png)
 

My guess is that it may be something with the configuration? but to be honest I can't say something for sure. If you guys can give me access to the matomo account I could dig a bit into it.

Anyways, removed the "module" as was not needed and passed it to the head.

neteler commented on themes/grass/static/plugins/tracking/matomo.js:
```diff
@@ -0,0 +1,10 @@
+var _paq = window._paq = window._paq || [];
+_paq.push(['trackPageView']);
```
```suggestion
/* Call disableCookies before calling trackPageView */
_paq.push(['disableCookies']);
_paq.push(['trackPageView']);
```

Using `disableCookies`, see https://matomo.org/faq/general/faq_157/

neteler commented:
> If you guys can give me access to the matomo account I could dig a bit into it.

This is managed by @lucadelu .

neteler commented:
> Anyways, removed the "module" as was not needed and passed it to the head. Don't think those are causing the issue, but don't heart.

I have deployed you latest version on https://grass.osgeo.org. The module stuff is gone but still nothing arriving at matomo...

nobeeakon commented:
Hi @neteler , could you please try again? I have updated the `siteId` to match what was in one of the commits [here](https://github.com/OSGeo/grass-website/pull/338/commits/e48c0a1b8eb97138145968d1d1fc54a3f62f1a60#diff-eb03067982c11d448d3640ad2209238936e9fa619da2e002cc1c471084ab505eL66). Have no way of knowing if that is the correct one, but may be. Also added the `disabledCookies`

lucadelu commented:
Now something is working

![Screenshot from 2022-11-19 06-53-02](https://user-images.githubusercontent.com/178741/202836822-19b1791d-b01f-4bce-be05-85dd280b1983.png)

neteler commented:
Thanks so much @nobeeakon (and @lucadelu). Matomo now works and does not leak any personal info, just the big picture which is enough for us.

E.g.: Page transitions within the site (users "walking" from here to there within the pages):

![image](https://user-images.githubusercontent.com/1295172/202850157-af48ccaa-db17-43dd-abca-01331eea152d.png)

neteler commented on content/about/privacy.md:
```diff
@@ -0,0 +1,24 @@
+
+**Matomo**
+
+This website uses the open source web analytics service [Matomo](https://matomo.org/). Matomo uses technologies that enable the recognition of the user across pages for the analysis of user behaviour (e.g. cookies or device fingerprinting). The information collected by Matomo about the use of this website is stored on our server. The IP address is anonymised before storage.
```
```suggestion
This website uses the open source web analytics service [Matomo](https://matomo.org/). Matomo uses technologies that enable the recognition of the user across pages for the analysis of user behaviour. In our case no cookies are collected. The information collected by Matomo about the use of this website is stored on our server. The IP address is anonymised before storage.
```

neteler commented on content/about/privacy.md:
```diff
@@ -0,0 +1,24 @@
+**Matomo**
+
+This website uses the open source web analytics service [Matomo](https://matomo.org/). Matomo uses technologies that enable the recognition of the user across pages for the analysis of user behaviour (e.g. cookies or device fingerprinting). The information collected by Matomo about the use of this website is stored on our server. The IP address is anonymised before storage.
+With the help of Matomo, we are able to collect and analyse data about the use of our website by website visitors. This enables us to find out, among other things, when which page views were made and from which region they come. We also collect various log files (e.g. IP address, referrer, browsers and operating systems used) and can measure whether our website visitors perform certain actions (e ...
```
```suggestion
With the help of Matomo, we are able to collect and analyse data about the use of our website by website visitors. This enables us to find out, among other things, when which page views were made and from which region they come. We also collect various log files (e.g. IP address, referrer, browsers and operating systems used) and can measure whether our website visitors perform certain actions (e.g. clicks, downloads, etc.).
```

neteler commented on content/about/privacy.md:
```diff
@@ -0,0 +1,24 @@
+
+We use IP anonymisation for the analysis with Matomo. This means that your IP address is shortened before analysis so that it can no longer be clearly assigned to you.
+
+If you do not agree to the storage and use of your data, you can deactivate the storage and use here. In this case, an opt-out cookie will be stored in your browser, which prevents Matomo from storing usage data. If you delete your cookies, this will have the effect that the Matomo opt-out cookie will also be deleted. The opt-out must be reactivated when you visit our site again.
```
```suggestion
```

I assume that this section isn't relevant any more? Just to avoid confusion.
(@lucadelu what do you think?)

neteler commented on content/about/privacy.md:
```diff
@@ -0,0 +1,24 @@
+
+If you do not agree to the storage and use of your data, you can deactivate the storage and use here. In this case, an opt-out cookie will be stored in your browser, which prevents Matomo from storing usage data. If you delete your cookies, this will have the effect that the Matomo opt-out cookie will also be deleted. The opt-out must be reactivated when you visit our site again.
+
+If you do not agree with the storage and use of your data, you can disable this feature here. In this case, an opt-out cookie will be stored in your browser to prevent Matomo from storing your usage data. If you delete your cookies, this will mean that the opt-out cookie will also be deleted. You will then need to reactivate it when you return to our site if you wish your activity not to be track ...
```
```suggestion
```

(the same)

veroandreo commented:
> Thanks so much @nobeeakon (and @lucadelu). Matomo now works and does not leak any personal info, just the big picture which is enough for us.
> 
> E.g.: Page transitions within the site (users "walking" from here to there within the pages):
> 
> ![image](https://user-images.githubusercontent.com/1295172/202850157-af48ccaa-db17-43dd-abca-01331eea152d.png)
> 

This is really cool, thanks @nobeeakon, @neteler and @lucadelu !! 🤓

neteler commented:
(FYI - I have removed the local pathing on the server, i.e. after merge this code should go in smoothly. Also means that matomo stats stopped for now).

veroandreo commented:
If ready, I'd suggest to merge then :)

nobeeakon commented:
think is ready to merge

neteler commented:
Thanks again @nobeeakon - merged!
